src/stdune/path.mli

```diff
@@ -1,7 +1,26 @@
+module Relative : sig
```

Comment:
Could you write a comment describing the set of valid values for this type?

Are we going for something like this:

```
A [Relative.t] is an arbitrary string that doesn't start with a "/". That is, "foo///bar" and "foo/bar" are distinct valid relative paths (even though going through them has the same effect). "foo/../foo/bar" is also distinct, and in fact for external paths its effect can be different.
```

It feels that many functions have weird behavior or fail particularly often on these:

- Everything that works with `basename` has weird behavior or should fail on files that end  in "..".
- `is_suffix` is not as meaningful (I guess that's expected)
- It's not clear what `parent` is supposed to mean: appending ".." and removing last component both make sense, but do different things

Maybe this module shouldn't implement Path_intf at all, or Path_intf should be split in two: one for paths under some root (local,  and absolute are both like that), and the relative ones like these.
```

Reply:
I agree that `Path_intf` should probably be split. I already removed a bunch of functions from `Path.Relative`, but I think you're right that we should be as minimal as possible.

Reply:
I've created `Path_intf.Local` which currently defines `parent` and `parent_exn`. I think those were the only objectionable functions remaining in `Path.Relative`.